packages/financial-templates-lib/src/price-feed/USPAC10GPriceFeed.ts

```diff
@@ -0,0 +1,364 @@
+      for (const q of quotes) {
+        price = price.addUnsafe(q);
+      }
+    }
+    price = price.divUnsafe(FixedNumber.from(this.basketItems));
+
+    return this.toBN(
+      price
+        .mulUnsafe(FixedNumber.from(this.toBN("10").pow(this.toBN(this.priceFeedDecimals)).toString()))
+        .round()
+        .toString()
+        .split(".")[0]
+    );
+  }
+
+  public getCurrentPrice(): BN | null {
+    return this.currentPrice;
+  }
+
+  public async getHistoricalPrice(time: number, ancillaryData: string, verbose = false): Promise<BN> {
+    const symbols = await this.fetchSymbolArray(verbose);
+
+    const symbolsHistories: RapidAPIHistoricalResponse[] = [];
+
+    for (let i = 0; i < symbols.length; i++) {
+      const symbol = symbols[i];
+      const url = `https://mboum-finance.p.rapidapi.com/hi/history?symbol=${symbol}&interval=${this.interval}&diffandsplits=false`;
+      const options = {
+        method: "GET",
+        headers: {
+          "x-rapidapi-host": "mboum-finance.p.rapidapi.com",
+          "x-rapidapi-key": this.rapidApiKey,
+        },
+      };
+      const response = (await this.networker.getJson(url, options)) as RapidAPIHistoricalResponse | null | undefined;
+
+      if (response == undefined || response.items == undefined)
+        throw new Error(`🚨Could not get quote results from url ${url}: ${JSON.stringify(response)}`);
+
+      symbolsHistories.push(response);
+    }
+
+    if (symbolsHistories.length !== this.basketItems) throw new Error(`🚨Could not get ${this.basketItems} quotes`);
+
+    const prices = [];
+    for (let i = 0; i < symbolsHistories.length; i++) {
+      const items = symbolsHistories[i].items;
+      const availableTimestamp = Object.keys(items).reverse();
+      for (let j = 0; j < availableTimestamp.length; j++) {
+        const ts = availableTimestamp[j];
+        if (Number(ts) <= time) {
+          prices.push(items[ts].close);
+          break;
+        }
+      }
+    }
+
+    if (prices.length !== this.basketItems) throw new Error(`🚨Could not get and format ${this.basketItems} prices`);
+
+    let price = FixedNumber.from(0);
+    for (let i = 0; i < prices.length; i++) {
+      price = price.addUnsafe(FixedNumber.from(prices[i].toString()));
+    }
+    price = price.divUnsafe(FixedNumber.from(this.basketItems.toString()));
+
+    return this.toBN(
+      price
+        .mulUnsafe(FixedNumber.from(this.toBN("10").pow(this.toBN(this.priceFeedDecimals)).toString()))
+        .round()
+        .toString()
+        .split(".")[0]
+    );
+  }
+
+  private async fetchSymbolArray(verbose = false): Promise<string[]> {
+    const url = "https://spachero-spac-database.p.rapidapi.com/top10/";
+    const options = {
+      method: "GET",
+      headers: {
+        "x-rapidapi-host": "spachero-spac-database.p.rapidapi.com",
+        "x-rapidapi-key": this.rapidApiKey,
+      },
+      body: '{ period: "weekly", type: "common", sortby: "gainers" }',
```

Comment:
@VasilyGerrans I am getting the following error: `TypeError: Request with GET/HEAD method cannot have body`. When I comment out the body the response is returned but wanted to check if it works for you?

I added the config below to https://github.com/UMAprotocol/protocol/blob/master/packages/financial-templates-lib/src/price-feed/DefaultPriceFeedConfigs.ts#L1019 and included my rapidApiKey in my `PRICE_FEED_CONFIG` as an env variable.

`uSPAC10g: {
    type: "uSPAC10g",
    lookback: 7200
  }`

Reply:
> @VasilyGerrans I am getting the following error: `TypeError: Request with GET/HEAD method cannot have body`. When I comment out the body the response is returned but wanted to check if it works for you?
> 
> I added the config below to https://github.com/UMAprotocol/protocol/blob/master/packages/financial-templates-lib/src/price-feed/DefaultPriceFeedConfigs.ts#L1019 and included my rapidApiKey in my `PRICE_FEED_CONFIG` as an env variable.
> 
> `uSPAC10g: { type: "uSPAC10g", lookback: 7200 }`

Could you also share the exact command you use for testing?

Reply:
Here is the command I am running (I have tried used various timestamps):

`HARDHAT_NETWORK=mainnet ./src/local/getHistoricalPrice.js --network mainnet_mnemonic --identifier uSPAC10g --time 1673276770 --ancillaryData 0x`

Reply:
Hey @VasilyGerrans, following up to see how it is looking on your end when you run the script or you had any questions.